.github/workflows/documentation.yml

```diff
@@ -47,31 +91,31 @@ jobs:
       - name: rust-src
         run: rustup component add rust-src --toolchain nightly
 
-      # TODO: we could build this once and download onto each runner
-      # Build the `xtask` package using the latest commit, and copy the
-      # resulting binary to the `~/.cargo/bin/` directory. We do this to
-      # avoid having to rebuild different versions of the package for
-      # different tags if they do not fall on the same commit, and to
-      # make sure that we take advantage of any subsequent updates to
-      # the xtask which may have happened after a release was tagged.
+      # Checkout the tag we need to start building the docs
       - name: Checkout repository
         uses: actions/checkout@v4
+        if: ${{ github.event_name == 'workflow_dispatch' }}
         with:
           repository: esp-rs/esp-hal
-      - name: Build xtask
-        run: |
-          cargo build --release --package=xtask --features=deploy-docs
-          cp target/release/xtask ~/.cargo/bin/hal-xtask
+          ref: ${{ matrix.packages.tag }}
 
-      # Checkout the tag we need to start building the docs
+      # If running a release PR, we want to build docs for the latest commit on the released branch.
       - name: Checkout repository
         uses: actions/checkout@v4
+        if: ${{ github.event_name != 'workflow_dispatch' }}
         with:
           repository: esp-rs/esp-hal
-          ref: ${{ matrix.packages.tag }}
+
+      - name: Download xtask
+        uses: actions/download-artifact@v4
+        with:
+          name: xtask
+          path: ~/.cargo/bin/
 
       - name: Build documentation
-        run: hal-xtask build documentation --packages=${{ matrix.packages.name }} --base-url /projects/rust/
+        run: |
+          chmod +x ~/.cargo/bin/xtask
+          ~/.cargo/bin/xtask build documentation --packages=${{ matrix.packages.name }} --base-url ${{ fromJSON('["https://preview-docs.espressif.com/projects/rust/", "https://docs.espressif.com/projects/rust/"]')[github.event.inputs.server == 'production'] }}
```

Comment:
Is this required? It worked before with just the `/projects/rust/`?

Reply:
Ahhh, I see. It worked in the sense of relative URLs once it was deployed, but it had no idea where to find the current deployment. Thanks!